diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee46061..5d0e0f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,20 @@ SET(CMAKE_INSTALL_RPATH "$ORIGIN")
 
 project(fep3-sdk VERSION ${VERSION})
 
+message(STATUS "Including Conan build info")
+if ( EXISTS ${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo.cmake)
+    include(${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo.cmake)
+elseif ( EXISTS ${CMAKE_CURRENT_BINARY_DIR}/../conanbuildinfo.cmake)
+    include(${CMAKE_CURRENT_BINARY_DIR}/../conanbuildinfo.cmake)
+elseif ( EXISTS ${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo_multi.cmake)
+    include(${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo_multi.cmake)
+elseif ( EXISTS ${CMAKE_CURRENT_BINARY_DIR}/../conanbuildinfo_multi.cmake)
+    include(${CMAKE_CURRENT_BINARY_DIR}/../conanbuildinfo_multi.cmake)
+else()
+    message(FATAL_ERROR "Conan build info can't be found.")
+endif()
+conan_basic_setup(TARGETS NO_OUTPUT_DIRS)
+
 set(FEP_SDK_PKG_VERSION_MAJOR ${PROJECT_VERSION_MAJOR})
 set(FEP_SDK_PKG_VERSION_MINOR ${PROJECT_VERSION_MINOR})
 set(FEP_SDK_PKG_VERSION_PATCH ${PROJECT_VERSION_PATCH})
@@ -62,6 +76,7 @@ option(fep3_sdk_cmake_enable_documentation
 ################################################################################
 find_package(fep3_system REQUIRED)
 find_package(fep3_participant REQUIRED)
+find_package(clipp REQUIRED)
 
 set(examples_integrated ON)
 add_subdirectory(examples)
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 37f5893..49a4b1c 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -18,17 +18,17 @@ if (NOT examples_integrated)
     set_property(GLOBAL PROPERTY USE_FOLDERS true)
 
 endif()
-
+# disable all non-gcc9 compatible examples
 add_subdirectory(src/rpc_service_bus)
-add_subdirectory(src/easy_cpp_example_datajob)
-add_subdirectory(src/easy_core_example_element)
+#add_subdirectory(src/easy_cpp_example_datajob)
+#add_subdirectory(src/easy_core_example_element)
 add_subdirectory(src/logging_example)
-add_subdirectory(src/data_triggered_simulation)
-add_subdirectory(src/dynamic_array)
-add_subdirectory(src/minimal_participant)
+#add_subdirectory(src/data_triggered_simulation)
+#add_subdirectory(src/dynamic_array)
+#add_subdirectory(src/minimal_participant)
 add_subdirectory(src/component_plugin)
 add_subdirectory(src/async_waitset_example)
-add_subdirectory(src/multi_sender_for_one_signal_example)
+#add_subdirectory(src/multi_sender_for_one_signal_example)
 add_subdirectory(src/signal_mapping_example)
 add_subdirectory(src/fep_interface_example)
-add_subdirectory(src/default_job_example)
+#add_subdirectory(src/default_job_example)
